src/tree.ml

```diff
@@ -693,7 +703,7 @@ let rec compile : CM.t -> tree = fun ({clauses ; positions ; slot} as pats) ->
       (* Abstraction specialisation*)
       let abstraction =
         if List.for_all (fun (x, _) -> x <> TC.Abst) cons then None else
-        let var = Bindlib.new_var mkfree "tr" in
+        let var = Bindlib.new_var mkfree (make_fresh_varname ()) in
```

Comment:
Isn't make_fresh_varname duplicating Bindlib code @rlepigre ?
If you want variable names to start with "var0", you can probably do once an `ignore(Bindlib.new_var mkfree "var")` somewhere.

Reply:
If you have not already done it, read the Bindlib paper or documentation.

Reply:
It is for the display of decision trees, to be able to distinguish matched variables (the name collision of Bindlib works like a charm, but isn't visible).

Reply:
I don't get it, they should already be distinguished if there is a name collision.

Reply:
In the graphviz file, we see only "var" for all variables.  Adding the integers enables to *see* "var1", "var2" in the generated graphviz. It is only used for the sake of being more informative in the visible trees.

Reply:
But aren't the names correct otherwise? What do you mean by "more informative"?

Reply:
I guess two images is worth two thousand words in that case, compare
![flip_ok](https://user-images.githubusercontent.com/7418676/68989967-a7dcf080-084d-11ea-9836-a3f327f0dbd1.png)
with this "less informative" image
![flip_KO](https://user-images.githubusercontent.com/7418676/68989969-aa3f4a80-084d-11ea-90d2-5bbbc70ef1cc.png)

Reply:
which is still a bit confusing since the rhs uses `x` and `y`, but closedness constraints would use the `var1` and `var2` variables. And as you suggested @rlepigre, this is not essential, it is only used for prettyness.

Reply:
I see, but in that case why even give names to a variable? The only useful information here seems to be that we have an abstraction constructor. Why not just print the lambda?

Reply:
It _can_ have meaning in two cases:
- conditional nodes
![dx](https://user-images.githubusercontent.com/7418676/68995182-17bd9c00-088b-11ea-8546-37c1f7a47bdb.png)
- matching on variables
![vm](https://user-images.githubusercontent.com/7418676/68995187-1f7d4080-088b-11ea-86d0-84d02e79910e.png)

Reply:
I see. And what do these examples if your hack is removed? (I say it is a hack because that is definitely not the code you want to write with Bindlib. Actually, I wonder if using Bindlib variables here makes sense.)

Reply:
You would only see `var` instead of `var1`.

Reply:
Another way would be to have some kind of (imperative) hashtable in `tree_graphviz.ml` mapping each bindlib variable to a textual representation.

Reply:
Another way would be to use an (imperative) hashtable in `tree_graphviz` to map bindlib vars to strings

Reply:
OK. I think I would prefer that solution. Note that if you do that you'll need to use `Hashtbl.Make` to create custom hashtable module (using `Bindlib.eq_vars` and `Bindlib.hash_var`).

Reply:
Ah I am crippled by my `TC.Vari` which only use strings in `tree_types`, so it means that the modification has to be done in the variable name itself. Or the `TC.Vari` type must be changed. I'll see if I manage to get something interesting.

Reply:
OK. I never liked the fact that we use string there, but I don't see any other obvious solution (my understanding of this code is still not perfect).

Reply:
So now a hash table is used to map a numeric id to each variable (instead of storing this id into the name of the variable).